internal/cli/command/cluster/shell.go

```diff
@@ -136,32 +142,44 @@ func runShell(banzaiCli cli.Cli, options shellOptions, args []string) error {
 		commandArgs = args[1:]
 	}
 
-	retry, err := writeConfig(ctx, pipeline, orgId, id, tmpfile)
-	if err != nil {
-		if !interactive || !retry {
-			return errors.WrapIf(err, "writing kubeconfig")
-		}
+	if options.configFile == "" {
+		retry, err := writeConfig(ctx, pipeline, orgId, id, tmpfile)
+		if err != nil {
+			if !interactive || !retry {
+				return errors.WrapIf(err, "writing kubeconfig")
+			}
 
-		go func() {
-			for {
-				retry, err := writeConfig(ctx, pipeline, orgId, id, tmpfile)
-				if err != nil {
-					if !retry {
-						log.Fatalf("%v", err)
+			go func() {
+				for {
+					retry, err := writeConfig(ctx, pipeline, orgId, id, tmpfile)
+					if err != nil {
+						if !retry {
+							log.Fatalf("%v", err)
+						}
+						log.Warningf("cluster config is still not available. retrying in 30 seconds")
+					} else {
+						log.Infof("cluster config successfully written")
+						return
 					}
-					log.Warningf("cluster config is still not available. retrying in 30 seconds")
-				} else {
-					log.Infof("cluster config successfully written")
-					return
-				}
 
-				select {
-				case <-time.After(30 * time.Second):
-				case <-ctx.Done():
-					return
+					select {
+					case <-time.After(30 * time.Second):
+					case <-ctx.Done():
+						return
+					}
 				}
-			}
-		}()
+			}()
+		}
+	} else {
+		// write config file into temp file
```

Comment:
When is this temp file deleted?

Reply:
After the user exit from the shell:
https://github.com/banzaicloud/banzai-cli/blob/fcff888d5d8b8e44ecd0ae7bfe09620413cd2168/internal/cli/command/cluster/shell.go#L118

Reply:
got it, thanks!